core/src/main/java/hudson/Functions.java

```diff
@@ -665,7 +667,25 @@ public static boolean isAutoRefresh(HttpServletRequest request) {
     public static boolean isCollapsed(String paneId) {
     	return PaneStatusProperties.forCurrentUser().isCollapsed(paneId);
     }
-    
+
+    public static boolean useUserTimeZone() {
+        return TimeZoneProperty.forCurrentUser() != null;
+    }
+
+    @Nullable
+    public static String getUserTimeZone() {
+        return TimeZoneProperty.forCurrentUser();
+    }
+
+    public static String getUserTimeZonePostfix() {
+        if (!useUserTimeZone()) {
+            return "";
+        }
+
+        TimeZone tz = TimeZone.getTimeZone(getUserTimeZone());
```

Comment:
You know how you could avoid dealing with invalid values? Provide only valid values that can be used. Use [`ZoneId.getAvailableZoneIds()`](https://docs.oracle.com/javase/8/docs/api/java/time/ZoneId.html#getAvailableZoneIds--) to get a list of all the timezones that are possible currently. Then you can use an `<f:select/>` jelly component with a `doFillTimeZoneChoiceValues()` type function.

By using only valid timezones, now you won't fall back to anything. Let me know if you'd like a more concrete example or a PR to your PR or something.